### ISSUE TITLE
Semantic-release integration

### ISSUE THREAD

**Issue by austince** [naver/billboard.js#893]
Hey! Based on [our recent conversation](https://github.com/naver/billboard.js/issues/863#issuecomment-490337157), you seem like you might be open to a PR the integrates [semantic-release](https://github.com/semantic-release/semantic-release) to help you ship code more frequently and with fewer manual steps. Before I get too deep in the integration, I wanted to make sure this is something that would help you + the billboard.js community. 

If you're up for considering it, I think the best general approach would be:
- two distribution channels, `current` and `nightly` mapping to the branches:
  - `master` -> `current`
  - `nightly` -> `nightly`
- `current` will be released as it currently is
- `nightly` will be released as a [pre-release](https://github.com/semantic-release/evolution/blob/release-workflows/_drafts/release-workflows.md#pre-releases) with a name like `billboard.js@1.8.1-nightly.1` instead of the timestamp
  - I'd also consider changing `nightly` to `next` (and publishing with the `@next` tag) 
- using semantic-release's [changelog plugin ](https://github.com/semantic-release/changelog) to generate the changelog from the commits
- using semantic-release's [release-notes plugin ](https://github.com/semantic-release/release-notes-generator)to generate the release info from the changelog
- publish releases to [`npm`](https://github.com/semantic-release/npm) and to [`github`](https://github.com/semantic-release/github) 
- and, of course, still publish gh-pages from `master`

PR flow would then have to be: new code -> `nightly` -> `master`, if you're ok with that.

Again, I'm up for making all these changes - just want to get your feedback before really starting, but I think it could make your life a bit easier!

**Comment by netil** [comment]
### Documentation
The current changelog & release note generation are automated in somehow via commit logs.
```sh
yarn changelog yyyy-mm-dd
```
The full automation is great, but also it should have some manual intervention. Because some of the items  needs to be handled manually.

I haven't used these tools, so I'm not sure how the result will looks like.
Maybe I'm overly concerning about this~ :)

### Shift from nightly to @next
I was considering to adopt `@next` release channel, but the @next should be for RC builds rather than the nightly. I think is too excessive, If is published everyday. Don't you?

We don't have some "adopt/test period" before the official release yet, so to adopt `@next` need to adopt some short period for this.

### Nightly versioning
I adopted timestamp to give how `freshness` of the build.
Is not a big issue changing to incremental versioning. 

----

Basically, automation is the right direction. So, implement first and let's see if works as expected.
If something goes wrong, try figure out improvement on that moment.

**Comment by austince** [comment]
Glad to see you're on board with moving towards automation! Do you have any specific items that need manual handling so I can work around them? Definitely understand where your uncertainties are coming from. I totally agree that releasing nightly is excessive and think a trial/adopt period for `@next` would be a good thing for large changes. 

We can certainly try to create our own plugin/way to use the current changelog script, though if you're comfortable with the results of the official plugin (which uses [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog) under the hood), it would take away some maintenance overhead from this lib to use it and hopefully give you more time to keep making billboard.js great! And I'll get started on an initial cut in my spare cycles 🎊

**Comment by netil** [comment]
@austince thanks in advance for the contribution.

>Do you have any specific items that need manual handling so I can work around them?

Is not about for specific tag nor the items. Sometimes the commits logs needs to be merged or removed depending the issues.

Usually after generating the changelog, I do some manual review to arrange that, but if everything goes automatically, I'm concerning on not having the chances for intervening this.

So, it would be nice to have some human intervention(if it's needed) before the public publish.

**Comment by austince** [comment]
Ok, sounds great! That's plenty to go on.
